java/code/src/com/suse/manager/maintenance/MaintenanceManager.java

```diff
@@ -695,6 +696,57 @@ public boolean isActionInMaintenanceWindow(Action action, MaintenanceSchedule sc
         return false;
     }
 
+    private Collection<CalendarComponent> getDateFromSchedule(MaintenanceSchedule ms) {
+        return getScheduleEventsAtDate(new Date(), ms, parseCalendar(ms.getCalendarOpt().get()));
```

Comment:
`ms.getCalendarOpt().get()` is a potential NullPointerException

Reply:
Can we find a better name for this method?

Reply:
There is still the problem with the potential `NPEx`. If the schedule does not have a calendar, it's the same as if it had a calendar with no events, the code should check for something like that:


```java
        return ms.getCalendarOpt()
                .map(cal -> getScheduleEventsAtDate(new Date(), ms, parseCalendar(cal)))
                .orElse(Collections.emptyList());
```

Reply:
As mc and me said: there is a high risk of `NPEx` (schedule can exist without calendar, so the risk is pretty high). You can fix this using the code I pasted in my latest comment.